jobs/rocoto/coupled_ic.sh

```diff
@@ -53,18 +53,40 @@ fi
 
 # Setup ATM initial condition files
 cp -r $ORIGIN_ROOT/$CPL_ATMIC/$CDATE/$CDUMP/*  $ICSDIR/$CDATE/atmos/
+rc=$?
+if [[ $rc -ne 0 ]] ; then
```

Comment:
None of these will catch errors if a single file in the group is missing, only if they are all missing. Probably sufficient for now though.

Reply:
Fair enough, it did catch the current error on orion. Suggestions for updates to improve beyond this are welcome.

Reply:
The best way would be to iterate over the files individually after copy to make sure they are there. The slightly lazier way would be to count the files after copy to make sure the right number are there. Either is a bit annoying to maintain.

I think this is probably good enough for now, as the vast majority of instances will be the whole directory is missing/empty, and this will catch those. We can think about more robust checking as build out the new unified system.

Reply:
May I suggest one change.

Instead of exiting right after a failed copy, initialize a counter `err_count=0` at the beginning and increment a counter e.g.`err_count` with `$rc`.

Let the script proceed till after all the copies are done.

At the end if `err_count > 0`, exit with a message that something failed to copy.

That way, the user gets a full view of what else is going to be missing, in case they fix this error and move on.

Reply:
Good suggestion. This is how build_all does it.

Reply:
I updated the script based on your comments.  Let me know what you think.

Reply:
looks good.